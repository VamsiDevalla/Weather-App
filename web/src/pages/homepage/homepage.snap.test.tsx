import React from 'react';
import renderer from 'react-test-renderer';
import CoordinateForm from '../../components/coordinates-form/coordinate-form.component';
import Map from '../../components/map/map.component';
import Homepage from './homepage.component';

jest.mock('../../components/map/map.component', () => 'Map');
jest.mock('../../components/weather/weather.component', () => 'Weather');
jest.mock('../../components/coordinates-form/coordinate-form.component', () => 'CoordinateForm');

const mockNavigatorGeolocation = () => {
  const clearWatchMock = jest.fn();
  const getCurrentPositionMock = jest.fn().mockImplementation(success =>
    Promise.resolve(
      success({
        coords: {
          latitude: 51.1,
          longitude: 45.3,
        },
      }),
    ),
  );
  const watchPositionMock = jest.fn();

  const geolocation = {
    clearWatch: clearWatchMock,
    getCurrentPosition: getCurrentPositionMock,
    watchPosition: watchPositionMock,
  };

  Object.defineProperty(global.navigator, 'geolocation', {
    value: geolocation,
  });

  Object.defineProperty(global.navigator, 'permissions', {
    query: jest.fn().mockImplementation(() => Promise.resolve({ state: 'granted' })),
  });

  return { clearWatchMock, getCurrentPositionMock, watchPositionMock };
};
describe('Homepage', () => {
  afterAll(() => jest.clearAllMocks());
  test('should render correctly', () => {
    const homepage = renderer.create(<Homepage />);
    expect(homepage.toJSON()).toMatchSnapshot();
    renderer.act(() => homepage.root.findByType(CoordinateForm).props.setCoordinates({ lat: 10, lng: 15 }));
    expect(homepage).toMatchSnapshot();
    renderer.act(() => homepage.root.findByType(CoordinateForm).props.setZoom(10));
    expect(homepage).toMatchSnapshot();
    renderer.act(() => homepage.root.findByType(Map).props.coordsHandler(10));
    expect(homepage).toMatchSnapshot();
    renderer.act(() => homepage.root.findByType(Map).props.zoomHandler(10));
    expect(homepage).toMatchSnapshot();
  });
  test('should render correctly on navigation permission', () => {
    mockNavigatorGeolocation();
    const homepage = renderer.create(<Homepage />);
    expect(homepage.toJSON()).toMatchSnapshot();
  });
});
