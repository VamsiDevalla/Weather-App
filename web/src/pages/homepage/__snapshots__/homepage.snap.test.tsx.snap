// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render correctly 1`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
  />
  <CoordinateForm
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={15}
  />
  <Map
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    coordsHandler={[Function]}
    zoom={15}
    zoomHandler={[Function]}
  />
</div>
`;

exports[`Homepage should render correctly 2`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
  />
  <CoordinateForm
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={15}
  />
  <Map
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
    coordsHandler={[Function]}
    zoom={15}
    zoomHandler={[Function]}
  />
</div>
`;

exports[`Homepage should render correctly 3`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
  />
  <CoordinateForm
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={10}
  />
  <Map
    coords={
      Object {
        "lat": 10,
        "lng": 15,
      }
    }
    coordsHandler={[Function]}
    zoom={10}
    zoomHandler={[Function]}
  />
</div>
`;

exports[`Homepage should render correctly 4`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={10}
  />
  <CoordinateForm
    coords={10}
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={10}
  />
  <Map
    coords={10}
    coordsHandler={[Function]}
    zoom={10}
    zoomHandler={[Function]}
  />
</div>
`;

exports[`Homepage should render correctly 5`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={10}
  />
  <CoordinateForm
    coords={10}
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={10}
  />
  <Map
    coords={10}
    coordsHandler={[Function]}
    zoom={10}
    zoomHandler={[Function]}
  />
</div>
`;

exports[`Homepage should render correctly on navigation permission 1`] = `
<div
  className="homepage"
  data-testid="homepage-container"
>
  <Weather
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
  />
  <CoordinateForm
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    initCoords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    setCoordinates={[Function]}
    setZoom={[Function]}
    zoom={15}
  />
  <Map
    coords={
      Object {
        "lat": 39.887710232733696,
        "lng": -75.17592430114746,
      }
    }
    coordsHandler={[Function]}
    zoom={15}
    zoomHandler={[Function]}
  />
</div>
`;
